templates/task/cf.yml

```diff
@@ -0,0 +1,106 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "CloudFormation template that represents a task on Amazon ECS."
+Parameters:
+  TaskName:
+    Type: String
+  TaskCPU:
+    Type: String
+  TaskMemory:
+    Type: String
+  LogRetention:
+    Type: Number
+  ContainerImage:
+    Type: String
+  TaskRole:
+    Type: String
+  Command:
+    Type: String
+Conditions:
+  HasImage:
+    !Not [!Equals [!Ref ContainerImage, ""]]
+  UseDefaultTaskRole:
+    !Equals [!Ref TaskRole, ""]
+  HasCommand:
+    !Not [!Equals [!Ref Command, ""]]
+Resources:
+  TaskDefinition:
+    Condition: HasImage
```

Comment:
Dumb question: shouldn't there always be an image? or what's a usecase where this condition is false?

Reply:
When we need to build the image from Dockerfile, we first need to create the ECR repo for the task, push the image to the repo, and then use it in the task definition. 

When we are creating the ECR repo, we cannot push the image simultaneously, hence there's no image for the task definition to use, which is the case where the condition is false.

Reply:
OOoooohhh that makes sense! hmm hmm what do you think of adding a comment next to the `HasImage` condition explaining what you just said above?